packages/main/src/CardHeader.js

```diff
@@ -0,0 +1,302 @@
+
+import {
+	AVATAR_TOOLTIP,
+	ARIA_ROLEDESCRIPTION_CARD_HEADER,
+	ARIA_ROLEDESCRIPTION_INTERACTIVE_CARD_HEADER,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Styles
+import cardHeaderCss from "./generated/themes/CardHeader.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-card-header",
+	languageAware: true,
+	managedSlots: true,
+	slots: /** @lends sap.ui.webcomponents.main.CardHeader.prototype */ {
+		avatar: {
+			type: HTMLElement,
+		},
+
+		/**
+		 * Defines an action, displayed in the right most part of the header.
+		 * <br><br>
+		 * <b>Note:</b> If set, the <code>status</code> text will not be displayed,
+		 * you can either have <code>action</code>, or <code>status</code>.
+		 * @type {HTMLElement[]}
+		 * @slot
+		 * @public
+		 * @since 1.0.0-rc.15
+		 */
+		action: {
+			type: HTMLElement,
+		},
+	},
+	properties: /** @lends sap.ui.webcomponents.main.CardHeader.prototype */ {
+
+		/**
+		 * Defines the title displayed in the component header.
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 * @since 1.0.0-rc.15
+		 */
+		titleText: {
+			type: String,
+		},
+
+		/**
+		 * Defines the subtitle displayed in the component header.
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 * @since 1.0.0-rc.15
+		 */
+		subtitleText: {
+			type: String,
+		},
+
+		/**
+		 * Defines the status displayed in the component header.
+		 * <br><br>
+		 * <b>Note:</b> If the <code>action</code> slot is set, the <code>status</code> will not be displayed,
+		 * you can either have <code>action</code>, or <code>status</code>.
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 * @since 1.0.0-rc.15
+		 */
+		status: {
+			type: String,
+		},
+
+		/**
+		 * Defines if the component header would be interactive,
+		 * e.g gets hover effect, gets focused and <code>headerPress</code> event is fired, when it is pressed.
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 * @since 1.0.0-rc.15
+		 */
+		headerInteractive: {
+			type: Boolean,
+		},
+
+		/**
+		 * Defines the aria-label attribute for the component
+		 *
+		 * @type {String}
+		 * @since 1.0.0-rc.15
+		 * @private
+		 * @defaultvalue ""
+		 */
+		ariaLabel: {
+			type: String,
+		},
+
+		/**
+		 * Receives id(or many ids) of the elements that label the component
+		 *
+		 * @type {String}
+		 * @defaultvalue ""
+		 * @private
+		 * @since 1.0.0-rc.15
+		 */
+		ariaLabelledby: {
+			type: String,
+			defaultValue: "",
+		},
+
+		_headerActive: {
+			type: Boolean,
+			noAttribute: true,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.main.CardHeader.prototype */ {
+
+		/**
+		 * Fired when the header is activated
+		 * by mouse/tap or by using the Enter or Space key.
+		 * <br><br>
+		 * <b>Note:</b> The event would be fired only if the <code>headerInteractive</code> property is set to true.
+		 * @event sap.ui.webcomponents.main.CarHeader#header-click
+		 * @public
+		 * @since 1.0.0-rc.15
+		 */
+		"header-click": {},
```

Comment:
This should not be called "header-click" anymore right, previously the event used to be fired from the Card, and "header-click" makes sense. So this should become just "click" within the CardHeader.
And if we want we can continue firing "header-click", but from the Card, this  we can do in future, no need to address it.